src/main/kotlin/io/github/yuutoproject/yuutobot/commands/Minigame.kt

```diff
@@ -0,0 +1,122 @@
+/*
+ * Open source bot built by and for the Camp Buddy Discord Fan Server.
+ *     Copyright (C) 2020  Kyuuto-devs
+ *
+ *     This program is free software: you can redistribute it and/or modify
+ *     it under the terms of the GNU General Public License as published by
+ *     the Free Software Foundation, either version 3 of the License, or
+ *     (at your option) any later version.
+ *
+ *     This program is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU General Public License for more details.
+ *
+ *     You should have received a copy of the GNU General Public License
+ *     along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package io.github.yuutoproject.yuutobot.commands
+
+import com.fasterxml.jackson.core.type.TypeReference
+import io.github.yuutoproject.yuutobot.commands.base.AbstractCommand
+import io.github.yuutoproject.yuutobot.commands.base.CommandCategory
+import io.github.yuutoproject.yuutobot.commands.minigame.MinigameInstance
+import io.github.yuutoproject.yuutobot.commands.minigame.MinigameListener
+import io.github.yuutoproject.yuutobot.objects.Question
+import io.github.yuutoproject.yuutobot.utils.jackson
+import net.dv8tion.jda.api.events.message.guild.GuildMessageReceivedEvent
+import net.dv8tion.jda.api.events.message.guild.react.GuildMessageReactionAddEvent
+import net.dv8tion.jda.api.events.message.guild.react.GuildMessageReactionRemoveEvent
+
+class Minigame : AbstractCommand(
+    "minigame",
+    CommandCategory.GAME,
+    "Play a fun quiz with your friends!",
+    "Run `minigame` to begin a new game, and react within the countdown to join.\nRun `minigame skip` to skip a question you do not wish to answer."
+) {
+    // String is the channel ID
+    private var minigames = mutableMapOf<Long, MinigameInstance>()
+    private val listener = MinigameListener(this)
+    private val questions: List<Question>
+
+    init {
+        val json = jackson.readTree(this.javaClass.getResource("/minigame.json"))
+        questions = jackson.readValue(json.traverse(), object : TypeReference<List<Question>>() {})
+    }
+
+    override fun run(args: MutableList<String>, event: GuildMessageReceivedEvent) {
+        val id = event.channel.idLong
+
+        // Handling for when a game is already in progress
+        val minigame = minigames[id]
+        if (minigame != null) {
+            // If a user attempts to skip a question
+            if (args.getOrNull(0) == "skip") {
+                if (!minigame.players.contains(event.author.idLong)) {
+                    event.channel.sendMessage("You can't skip a question if you aren't in the game.").queue()
```

Comment:
```suggestion
                    event.channel.sendMessage("You can't skip a question if you aren't in the game!").queue()
```
Would be better with an exclamation mark